swarm/src/lib.rs

```diff
@@ -1083,6 +1077,15 @@ where
                     addrs.dedup();
                     addrs
                 };
+
+                // If address translation yielded nothing, broacast the original candidate address.
+                if translated_addresses.is_empty() {
+                    self.behaviour
+                        .on_swarm_event(FromSwarm::NewExternalAddrCandidate(
+                            NewExternalAddrCandidate { addr: &addr },
+                        ));
```

Comment:
I am contemplating whether we should early return with a `return None` here. The loop below is a no-op if we are in here but not having any control flow in here makes it look like we want the below to happen. Thoughts?

Reply:
I was overthinking it before I decided on this. I thought about `push`ing into the `Vec` or wrapping the loop in `else`, but yeah, the solution I implemented isn't that optimal I guess.

I'll just wrap the loop in `else`. Let me know if that's good.